### DISCUSSION TITLE
OTC Framework is interested in joining Commonhaus

### DISCUSSION THREAD

**Discussion by frank-a-otc** [commonhaus/foundation#190] [Joining Commonhaus]
### Project information

- **Project name**: OTC Framework
- **Project website**: https://otcframework.org/
- **Code repository**: https://github.com/frank-a-otc/otcframework
- **License**: Apache 2.0

### Do you have authority to represent this project?

- [X] I own or maintain this project

### Describe your project

I believe the OTC Framework is a good candidate because it is a very productive data-transformation Java source-code-generator for software developers to facilitate rapid development released under the Apache 2.0 open source license.

Be it data-transformations between different data-structures, or data-formats such as XML, JSON, YAML or for integrating with various microservices using REST / SOAP protocols, or with databases or UI applications, this framework is a great solution. This framework thrives in simplicity and performance with unique features to handle every kind of complex object-to-object data-transformation mappings, extremely quick up-and-running transformations, requiring very low developer effort, ease-of-use and delivering the best performance. The developers do not have to write any code - not even interfaces or annotations, or any lengthy configurations manually. All a developer has to do is, just create the OTC-script with just a few clicks on a intuitive OTC Script Editor UI and subsequently generate all the required data-transformation source code just on the click of a button. 

This open source project needs support and visibility to reach bigger audience.

Please visit [https://otcframework.org](https://otcframework.org) for more information.

### Things that must be done (bootstrap phase)

- [x] **Review the bylaws**, especially sections about [projects](https://www.commonhaus.org/bylaws/cf-membership.html#projects), [decision making](https://www.commonhaus.org/bylaws/decision-making.html), and [voting for significant decisions](https://www.commonhaus.org/bylaws/cf-council.html#voting).
- [x] **Read the [bootstrapping agreement](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md)**.
- [x] If all is well, and you agree to participate in the bootstrapping phase to help us build this thing, then **create a PR** with the following:

    1. A filled out copy of the bootstrapping agreement. You can make a copy of [bootstrapping.md](https://github.com/commonhaus/foundation/blob/main/agreements/bootstrapping/bootstrapping.md) file and check that in, or checkin a filled out [pdf](https://github.com/commonhaus/foundation/releases/download/final-draft/bootstrapping.pdf). Please follow naming conventions.
    2. An updated [CONTACTS.yaml](https://github.com/commonhaus/foundation/blob/main/CONTACTS.yaml) that names your project representative in the EGC section
    3. An updated [PROJECTS.yaml](https://github.com/commonhaus/foundation/blob/main/PROJECTS.yaml) that provides details / describes your project
    4. An updated [TRADEMARKS.md](https://github.com/commonhaus/foundation/blob/main/TRADEMARKS.md) that adds your project name and project logo (if you have one). Please include the footnote reference, as we are using your project name and logo with your permission until we have a formal agreement in place and resources are transferred.

    The commit should be "Signed-off-by" your GH login (`git commit -s -m "<project> bootstrapping agreement, yyyy-mm-dd"`)

**Comment by frank-a-otc** [comment]
@ebullient - Created [OTC Framework bootstrapping agreement, 2024-08-21 #192](https://github.com/commonhaus/foundation/pull/192)

**Comment by ebullient** [comment]
Sorry for the delay, Frank. 

Closing this discussion (for now) as resolved by #192. 
I hope we can revisit this again in April, when we've exited our bootstrapping phase.
